File: 11498.cpp
#include <iostream>

using namespace std;

int main() {
  while (true) {
    int tc;
    cin >> tc; 
    if (tc == 0) break;
    int dx, dy;
    cin >> dx >> dy;

    int x, y;
    for (int i = 0; i < tc; i++) {
      cin >> x >> y;
      if (x == dx or y == dy) cout << "divisa\n";
      else if (x > dx and y > dy) cout << "NE\n";
      else if (x < dx and y > dy) cout << "NO\n";
      else if (x > dx and y < dy) cout << "SE\n";
      else if (x < dx and y < dy) cout << "SO\n";
    }
  }
  return 0;
}

File: 11498.py
while True:
    t = int(input())
    if t == 0:
        break

    dx, dy = map(int, input().split())

    for _ in range(0, t):
        cx, cy = map(int, input().split())
        if dx == cx or dy == cy: print('divisa')
        elif cx > dx and cy > dy: print('NE')
        elif cx > dx and cy < dy: print('SE')
        elif cx < dx and cy > dy: print('NO')
        elif cx < dx and cy < dy: print('SO')
